src/api/spec/controllers/webui/download_on_demand_controller_spec.rb

```diff
@@ -0,0 +1,165 @@
+      it { expect(DownloadRepository.where(dod_parameters[:download_repository])).not_to exist }
+    end
+
+    context "valid requests" do
+      before do
+        login(admin_user)
+        post :create, dod_parameters
+      end
+
+      it { is_expected.to redirect_to(project_repositories_path(project)) }
+      it { expect(flash[:notice]).to eq("Successfully created Download on Demand") }
+      it { expect(assigns(:download_on_demand)).to be_kind_of(DownloadRepository) }
+      it { expect(DownloadRepository.where(dod_parameters[:download_repository])).to exist }
+    end
+
+    context "invalid parameter (architecture)" do
+      before do
+        dod_parameters[:download_repository][:arch] = ""
+        login(admin_user)
+        post :create, dod_parameters
+      end
+
+      it { is_expected.to redirect_to(root_path) }
+      it { expect(flash[:error]).to eq("Download on Demand can't be created: Arch can't be blank") }
+      it { expect(assigns(:download_on_demand)).to be_kind_of(DownloadRepository) }
+      it { expect(DownloadRepository.where(dod_parameters[:download_repository])).not_to exist }
+    end
+  end
+
+  describe 'DELETE destroy' do
+    let(:dod_repository) { create(:download_repository) }
+
+    before do
+      repository.download_repositories << dod_repository
+    end
+
+    it { is_expected.to use_before_action(:set_project) }
+
+    context "for non-admin users" do
+      before do
+        login(create(:confirmed_user))
+        delete :destroy, id: dod_repository.id, project: project.name
+      end
+
+      it { is_expected.to redirect_to(root_path) }
+      it { expect(flash[:error]).to eq("Sorry, you are not authorized to delete this DownloadRepository.") }
+      it { expect(DownloadRepository.where(id: dod_repository.id)).to exist }
+    end
+
+    context "valid requests" do
+      before do
+        login(admin_user)
+        delete :destroy, id: dod_repository.id, project: project.name
+      end
+
+      it { is_expected.to redirect_to(project_repositories_path(project)) }
+      it { expect(flash[:notice]).to eq("Successfully removed Download on Demand") }
+      it { expect(DownloadRepository.where(id: dod_repository.id)).not_to exist }
+    end
+
+    context "invalid requests" do
+      skip("Please add some tests:-)")
+    end
+  end
+
+  describe "POST update" do
+    let(:dod_repository) { create(:download_repository) }
+
+    before do
+      repository.download_repositories << dod_repository
+    end
+
+    it { is_expected.to use_before_action(:set_project) }
+
+    context "for non-admin users" do
+      before do
+        login(create(:confirmed_user))
+        dod_parameters[:id] = dod_repository.id
+        dod_parameters[:download_repository][:arch] = "s390x"
+
+        post :update, dod_parameters
+      end
+
+      it { is_expected.to redirect_to(root_path) }
+      it { expect(flash[:error]).to eq("Sorry, you are not authorized to update this DownloadRepository.") }
```

Comment:
But the authorization for that controller is handled by pundit. No require_admin filter involved here.